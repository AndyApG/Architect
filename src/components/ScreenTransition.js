import React from 'react';
import PropTypes from 'prop-types';
import { Transition } from 'react-transition-group';
import { animation } from 'network-canvas-ui';
import anime from 'animejs';

const fadeOut = {
  opacity: [1, 0],
  duration: animation.duration.fast,
};

const expand = {
  scale: [0, 1],
  opacity: [0, 1],
};

const ScreenTransition = ({ children, timeout, ...props }) => (
  <Transition
    timeout={timeout}
    unmountOnExit
    appear
    onEnter={
      (el) => {
        el.setAttribute('style', `${el.getAttribute('style')}; transform-origin: 50vw 50vh;`);

        anime({
          targets: el,
          elasticity: 0,
          easing: 'easeInOutQuad',
          duration: timeout,
          ...expand,
        });
      }
    }
    onExit={
      (el) => {
        el.setAttribute('style', `${el.getAttribute('style')}; transform-origin: 50vw 50vh;`);

        anime({
          targets: el,
          elasticity: 0,
          easing: 'easeInOutQuad',
          duration: timeout,
          ...fadeOut,
        });
      }
    }
    {...props}
  >
    { children }
  </Transition>
);

ScreenTransition.propTypes = {
  children: PropTypes.any.isRequired,
  timeout: PropTypes.number,
};

ScreenTransition.defaultProps = {
  children: PropTypes.any.isRequired,
  timeout: animation.duration.slow,
};

export default ScreenTransition;
