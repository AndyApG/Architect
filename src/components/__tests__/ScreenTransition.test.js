/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import ScreenTransition from '../ScreenTransition';

jest.mock('../../utils/CSSVariables');

const mockProps = {
};

describe('<ScreenTransition />', () => {
  it('can render', () => {
    const component = shallow(<ScreenTransition {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
