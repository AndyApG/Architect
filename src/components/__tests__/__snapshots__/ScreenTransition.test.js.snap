// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenTransition /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScreenTransition
    timeout={1000}
>
    [Function]
</ScreenTransition>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "appear": false,
      "children": [Function],
      "enter": true,
      "exit": true,
      "in": false,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "timeout": 1000,
      "unmountOnExit": true,
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "appear": false,
        "children": [Function],
        "enter": true,
        "exit": true,
        "in": false,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "timeout": 1000,
        "unmountOnExit": true,
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
